tools/src/test/java/org/apache/kafka/tools/consumer/group/DeleteConsumerGroupsTest.java

```diff
@@ -17,279 +17,374 @@
-            "The group did not initialize as expected.");
-
-        executor.shutdown();
-
-        TestUtils.waitForCondition(
-            () -> Objects.equals(service.collectGroupState(GROUP).state, "Empty"),
-            "The group did not become empty as expected.");
-
-        cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--delete", "--group", GROUP, "--group", missingGroup};
+    @ClusterTemplate("generator")
+    public void testDeleteWithMixOfSuccessAndError() throws Exception {
+        for (GroupProtocol groupProtocol : cluster.supportedGroupProtocols()) {
+            String groupId = composeGroupId(groupProtocol);
+            String topicName = composeTopicName(groupProtocol);
+            String missingGroupId = composeMissingGroupId(groupProtocol);
+            String[] cgcArgs = new String[]{"--bootstrap-server", cluster.bootstrapServers(), "--delete", "--group", groupId};
+            try (
+                    AutoCloseable executor = consumerGroupClosable(groupProtocol, groupId, topicName);
+                    ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs)
+            ) {
+                TestUtils.waitForCondition(
+                        () -> service.listConsumerGroups().contains(groupId) && checkGroupState(service, groupId, STABLE),
+                        "The group did not initialize as expected.");
+
+                executor.close();
+
+                TestUtils.waitForCondition(
+                        () -> checkGroupState(service, groupId, EMPTY),
+                        "The group did not become empty as expected.");
+
+                cgcArgs = new String[]{"--bootstrap-server", cluster.bootstrapServers(), "--delete", "--group", groupId, "--group", missingGroupId};
+
+                ConsumerGroupCommand.ConsumerGroupService service2 = getConsumerGroupService(cgcArgs);
+                Map<String, Throwable> result = service2.deleteGroups();
+                assertTrue(result.size() == 2 &&
+                                result.containsKey(groupId) && result.get(groupId) == null &&
+                                result.containsKey(missingGroupId) &&
+                                result.get(missingGroupId).getMessage().contains(Errors.GROUP_ID_NOT_FOUND.message()),
+                        "The consumer group deletion did not work as expected");
+            }
+        }
+    }
 
-        ConsumerGroupCommand.ConsumerGroupService service2 = getConsumerGroupService(cgcArgs);
+    @ClusterTemplate("generator")
+    public void testDeleteWithUnrecognizedNewConsumerOption() {
+        String[] cgcArgs = new String[]{"--new-consumer", "--bootstrap-server", cluster.bootstrapServers(), "--delete", "--group", getDummyGroupId()};
+        assertThrows(OptionException.class, () -> getConsumerGroupService(cgcArgs));
```

Comment:
We expect the option parse get failed, so it would be more suitable to test `ConsumerGroupCommandOptions.fromArgs` directly